monkey/monkey_island/cc/next_ui/src/app/(protected)/plugins/available/AvailablePluginFilters.tsx

```diff
@@ -0,0 +1,86 @@
+import Grid from '@mui/material/Grid';
+import React, { useEffect, useMemo, useState } from 'react';
+import SearchFilter from '@/app/(protected)/plugins/_lib/filters/SearchFilter';
+import TypeFilter from '@/app/(protected)/plugins/_lib/filters/TypeFilter';
+import { useGetAvailablePluginsQuery } from '@/redux/features/api/agentPlugins/agentPluginEndpoints';
+import {
+    generatePluginsTableRows,
+    PluginRow
+} from '@/app/(protected)/plugins/_lib/PluginTable';
+import _ from 'lodash';
+import InstalledPluginFilter from '@/app/(protected)/plugins/_lib/filters/InstalledPluginFilter';
+
+type AvailablePluginFiltersProps = {
+    setDisplayedRowsCallback: (rows: PluginRow[]) => void;
+    setIsFilteringCallback: (isFiltering: boolean) => void;
+};
+
+export type FilterProps = {
+    setFiltersCallback: (filters: any) => void;
+};
+
+export const defaultSearchableColumns = [
+    'name',
+    'pluginType',
+    'version',
+    'author'
+];
+
+const AvailablePluginFilters = (props: AvailablePluginFiltersProps) => {
```

Comment:
This component seems pretty complex. Can we make it pure? Perhaps instead of the product being the filtered rows, we can make it an aggregated filter function?

Reply:
My idea was to decouple the page from filtering. If the filters component produces filters, then the parent component is left responsible for applying them. I can try to put more love into this and simplify

Reply:
I've looked more into it, but couldn't come up with any good alternatives. The point of this component is to describe the filtering logic and display of available plugins table. At the end of the day some component has to apply the filters to filter the rows.

Reply:
The purpose of this component is to describe the available plugin filtering logic and available plugin filter display. It needs to know about available plugins to filter them. It could be done in the parent, but I don't see what we'd gain. We'd trade saving one line of `const { data: availablePlugins } = useGetAvailablePluginsQuery();` for having worse separation of concerns and worse readability.

Reply:
I'm not too picky about this, but this is what I see: the component seems a bit brittle to me. This is because the component needs to know exactly which filters it has and how to position them, and therefore is tightly coupled to those components. That is one concern of this component. However, there is another concern, which doesn't care about how filters are being displayed, and that is the logic that applies the filters. It doesn't need to be coupled to the display at all.